hydra-node/test/Hydra/Model.hs

```diff
@@ -696,6 +743,33 @@ performClose party = do
       err@CommandFailed{} -> error $ show err
       _ -> False
 
+performFanout :: (MonadThrow m, MonadAsync m, MonadDelay m, MonadTimer m) => Party -> RunMonad m UTxO
+performFanout party = do
+  nodes <- gets nodes
+  let thisNode = nodes ! party
+  waitForClosed thisNode
+  party `sendsInput` Input.Fanout
+
+  lift $
+    waitUntilMatch (toList nodes) $ \case
+      HeadIsFinalized{} -> True
+      err@CommandFailed{} -> error $ show err
+      _ -> False
+
+  outputs <- lift $ serverOutputs thisNode
+  case find headIsFinalized outputs of
+    Just HeadIsFinalized{utxo} -> pure utxo
+    _ -> failure "Failed to perform Fanout"
```

Comment:
Should: not look twice for `HeadIsFinalized`, but already parse `utxo` from the first `waitUntilMatch` (might need to enhance this function)